packages/sanity/src/core/comments/utils/transform-children/linkMiddleware.ts

```diff
@@ -0,0 +1,30 @@
+import {createElement, type ReactNode} from 'react'
+
+import {type Middleware} from './types'
+
+const URL_REGEX = /\b(https?:\/\/[^\s,()]+(?:\.[^\s,()]+)*|www\.[^\s,()]+\.[^\s,()]{2,})\b/g
+
+function createLinkElement(url: string): ReactNode {
+  const href = url.startsWith('http') ? url : `https://${url}`
+
+  return createElement('a', {href, target: '_blank', rel: 'noopener noreferrer', key: url}, url)
```

Comment:
Stop propagation to avoid triggering click on the comment item when clicking the link?